bpo-44929: Fix enum string representation in the REPL

pablogsal opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- issue-number: [bpo-44929](https://bugs.python.org/issue44929) -->
https://bugs.python.org/issue44929
<!-- /issue-number -->

pablogsal committed:
Fix enum string representation in the REPL

ambv committed:
Add more tests for global int flag reprs

gpshead commented on Lib/enum.py:
```diff
@@ -1396,7 +1396,7 @@ def global_flag_repr(self):
     cls_name = self.__class__.__name__
     if self._name_ is None:
-        return "%x" % (module, cls_name, self._value_)
+        return "%s.%s(%x)" % (module, cls_name, self._value_)
```
Do any of the unittests test this particular format codepath?  none of the added tests have reprs that end with `(0x8)` or similar.

ambv replied:
Yeah, no, I couldn't quite figure out how to make `_name_` to be None. When we hear back from Ethan, he'll hopefully suggest a test for this, too. Worst case we can just add a test using `repr(re.RegexFlag(0))`.

ethanfurman replied:
https://bugs.python.org/msg399863

ethanfurman committed:
use last module if multi-module string

when an enum's `__module__` contains several module names, only use the
last one

ethanfurman reviewed (dismissed):
Fixed tests and global reprs.  Thank you.

ethanfurman commented on Lib/test/test_enum.py:
```diff
@@ -3224,6 +3244,26 @@ def test_repr(self):
+                repr(HeadlightsC(2**0 + 2**2 + 2**3)),
+                '%(m)s.LOW_BEAM_C|%(m)s.FOG_C' % {'m': SHORT_MODULE},
+                )
+
```
Still need test for `_name_ is None` branch.  I'll have that in shortly.

ethanfurman committed:
add completely-out-of-range flag test

ethanfurman committed:
remove extraneous blank lines
